source/docs/hardware-reference/talonfx/improving-performance-with-current-limits.rst

```diff
@@ -0,0 +1,215 @@
+
+.. note:: The below example ignores factors such as energy loss from heat or inefficiency.
+
+For example, if a motor is being applied with 50% dutycycle with :math:`80A`` of load (stator current), the resulting supply current will be :math:`40A`. If the motor has 100% dutycycle applied wth :math:`80A`` of stator current, the resulting supply current wll be :math:`80A`.
+
+.. figure:: images/stator-limit.png
+   :alt: 80a stator limit being applied
+
+   *80a torque limit taking affect*
+
+Stator limits apply the same limiting strategy used by supply limits, but are more effective in dealing with situations such as wheel slip or acceleration induced brownouts. The rationale behind it reducing acceleration induced brownouts is that initial acceleration is typically peak supply and stator current.
+
+If a robot acceleration event draws :math:`100A` of supply current, it will only do so for a very small portion of the entire acceleration event up to a maximum velocity. During the acceleration event, the motor is likely not being commanded with 100% dutycycle. Therefore, if :math:`100A` are being drawn at supply at :math:`6V`, then output stator is around :math:`200A`. A :math:`100A` stator limit would cap the supply current at `50A` while being more efficient at not limiting the entire robot velocity.
+
+How to Budget Limits
+--------------------
+
+Budgeting a current limit depends on the limit and your overall battery budget. When budgeting current limits for FRC, the most robust strategy is to run a robot in match-like conditions and observe battery draw and current draw through the course of the robot operation.
+
+Using Limits to Improve Battery Longevity
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+While supply limits can be estimated using battery datasheets and average mechanism current draw, the easiest method is to graph and reduce. By applying a conservative supply limit to mechanisms that can operate with minimal current draw, you can improve the performance of other mechanisms (e.g. capping your intake supply to increase the amount of current a swerve drivetrain can draw).
+
+1. Plot a mechanism supply currents throughout the match
+2. Reduce your supply current until your performance begins to suffer (unable to intake items, etc). Increase this value slightly for some comfort room.
+3. Repeat this for all mechanisms on a fresh battery each time. Time how long it takes before brownouts are regular and battery capacity has dropped.
+
+For example, a user may have the following mechanisms and supply limits.
+
+- x4 Kraken(s) on swerve drive - 60A supply
+- x4 Kraken(s) on swerve dzimuth - 20A supply
+- x1 Kraken(s) on elevator - 30A supply
+- x1 Kraken(s) on intake - 15A supply
+
+This would yield peak supply current of ~395A for a worst case scenario. This draw is extremely unlikely as peak supply is often extremely brief (for example, 60A on all 4 swerve drive motors will likely be for less than 5 seconds).
```

Comment:
Also mention how it's unlikely because you're not going to be accelerating all mechanisms at the same time.